mysql: Use a connection pool

rhansen opened the pull request:
cc @packardone

webzwo0i commented:
Cool! 

Do we still need the schedulePing stuff for mysql, or does the pool automatically opens/closes new connections?

https://github.com/ether/ueberDB/blob/master/test/test_mysql.js#L12 is not finishing anymore

webzwo0i commented:
https://github.com/webzwo0i/ueberDB/commit/7890c2ea21c5daa24d648562cc6103aaea9cbcf8 contains a small test for the pool

rhansen commented:
> Do we still need the schedulePing stuff for mysql, or does the pool automatically opens/closes new connections?

No, we don't need the pings anymore so I removed them. Thanks for pointing that out.

FYI, the pool automatically opens new connections when under the connection limit, and if a connection fails (e.g., fatal error) the pool will automatically remove it. The pool sends a ping each time a connection is reused to ensure that the connection is still healthy.

> https://github.com/ether/ueberDB/blob/6755ca6b62e31356e176de8f99261c71327605a9/test/test_mysql.js#L12 is not finishing anymore

Strange—I wonder why the test exited 0. It's working now.

> webzwo0i@7890c2e contains a small test for the pool.

Thanks! I took the second test, tweaked it a bit, and squashed it in.
